src/classes/FORM_ServiceGiftEntry.cls

```diff
@@ -297,4 +320,108 @@ public with sharing class FORM_ServiceGiftEntry {
         }
         return template;
     }
+
+    /******************************************************************************************************
+     * @description Private method for checking FLS & CRUD permissions for custom column headers for the
+     * the GE List View headers
+     * @param columnHeaders Custom Column Headers
+     * @param listName List View Name
+     * @return Instance of the ColumnHeaderData object
+    */
+    private static ColumnHeaderData getColumnHeaderDataWithPermissions(Custom_Column_Header__c[] columnHeaders,
+            String listName) {
+        String listNameDevName = listName == TEMPLATES ?
+                Schema.SObjectType.Form_Template__c.getName() :
+                Schema.SObjectType.DataImportBatch__c.getName();
+
+        ColumnHeaderData headerData;
+
+        if (columnHeaders.isEmpty()) {
+            //populate column headers with default headers
+            columnHeaders = populateDefaultColumnHeaders(listName);
+        }
+
+        FORM_PermissionValidator validator = new FORM_PermissionValidator(
+                new Set<FORM_PermissionValidator.AccessLevel>{FORM_PermissionValidator.AccessLevel.VALIDATE_READ});
+
+        if (validator.hasObjectPermissions(listNameDevName)) {
+            headerData = new ColumnHeaderData(columnHeaders, '');
+            List<String> flsErrors = new List<String>();
+            for (Custom_Column_Header__c header : columnHeaders) {
+                if (!validator.hasFieldPermissions(listNameDevName, header.Field_Api_Name__c)) {
+                    List<String> errors = new List<String>{header.Field_Api_Name__c};
+                    flsErrors.add(listNameDevName + ': (' + String.join(errors, ', ') + ')');
+                }
+            }
+            headerData.permissionErrorData += String.join(flsErrors, ' ,');
+        }
+        return headerData;
+    }
+
+    /**
+     * @description Method to populate default column headers for the Gift Entry list view tables
+     * @param listName
+     * @return {Custom_Column_Header__c}
+    */
+    private static Custom_Column_Header__c[] populateDefaultColumnHeaders(String listName) {
```

Comment:
Yeah, I mean, these are methods related to the list views that are on the Gift Entry tab and are not related to the form renderer, right?  It would make the most sense to have them in GE_GiftEntryController, but the app is probably due for a larger refactor of this class anyway - so we could probably do all the reorg/refactor of this class at once.

Reply:
Thumbs up from me for the refactor. I will add this to that list if it isn’t on there yet. Thanks guys!